src/builtins/impls/uuid.rs

```diff
@@ -12,10 +12,11 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use anyhow::{bail, Result};
+use anyhow::Result;
+use uuid::Uuid;
 
 /// Returns a new UUIDv4.
 #[tracing::instrument(name = "uuid.rfc4122", err)]
 pub fn rfc4122(k: String) -> Result<String> {
-    bail!("not implemented");
+    Ok(Uuid::new_v4().to_string())
```

Comment:
Just some more thoughts:

* I've implemented the hash/rng seeding strategy for myself to see. On every run we get the first random uuid to be the same.
* Next the requirement from OPA makes this stay the same: for any given `k` we must give out the same uuid

The combination of the two results with just a fancy hashing function.
I'll comment out everything here for the meanwhile and de-couple the UUID impl from the test infra PR, looks like we have to have some kind of global cache for all builtins to share, unless we come up with a smarter idea?

Reply:
A note from the OPA side:

> It means that we need to cache the value for every k, and that's how consistency is achieved.
> It also tells me to try to never generate UUIDs freely in OPA because they are costly - eventually it'll blow up my memory.

That' caching is per-evaluation, not overall. So within a single evaluation, the same `k` will yield the same uuid. for the next evaluation, it'll yield a different one, as the built-in context's cache is a fresh one. 

Caching that spans queries is only done for http.send responses.

Reply:
Hey @srenatus I was wondering if to tag someone from OPA, thanks for magically appearing! 🥇 
A few questions:

1. While I've read a bunch of historic issues on it, looks like the motivation for it was testing and the fact that one can never know how many times OPA will evaluate a block (so it has to be idempotent) -- did that actually materialize? (when I github search the uuid builtin -- looks like almost no one uses it, public in the open)
2. Does this effectively mean that If I ever wanted to generated different 2 uuid's within eval, I have to come up with 2 unique symbols (hardcoded) within the eval context?